src/sagemaker_containers/encoders.py

```diff
@@ -138,15 +140,20 @@ def encode(self, obj, content_type):
         Returns:
             object: encoded object.
         """
-        for encoder in self._encoders:
-            if content_type == encoder.content_type:
-                return encoder.encode(obj)
-        raise UnsupportedFormatError(content_type)
+        try:
+            return self._encoders_map[content_type].encode(obj)
+        except KeyError:
+            raise UnsupportedFormatError(content_type)
+
+
+default_encoder = DefaultEncoder()
```

Comment:
remove module var

Reply:
discussed offline. ok as is.

Reply:
@jesterhazy found the following in the google style guide:

[05/07/18 11:47 AM] Esterhazy, Jonathan: Global variables
link ▽ Avoid global variables.
Definition:
Variables that are declared at the module level.

Pros:
Occasionally useful.

Cons:
Has the potential to change module behavior during the import, because assignments to module-level variables are done when the module is imported.

Decision:
Avoid global variables in favor of class variables. Some exceptions are:

Default options for scripts.
Module-level constants. For example: PI = 3.14159. Constants should be named using all caps with underscores; see Naming below.
It is sometimes useful for globals to cache values needed or returned by functions.
If needed, globals should be made internal to the module and accessed through public module level functions; see Naming below.
[05/07/18 11:47 AM] Esterhazy, Jonathan: ttps://google.github.io/styleguide/pyguide.html?showone=Global_variables#Global_variables

Reply:
I am working in a solution outside this PR. I will add a linter as well, to better rule enforcement.
@jesterhazy is plans to work in a best practices guide for the team.